runtime/src/rent_collector.rs

```diff
@@ -32,6 +32,13 @@ impl Default for RentCollector {
     }
 }
 
+/// When rent is collected from an exempt account, rent_epoch is set to this
+/// value. The idea is to have a fixed, consistent value for rent_epoch for all accounts that do not collect rent.
+/// This enables us to get rid of the field completely.
+pub const RENT_EXEMPT_RENT_EPOCH: Epoch = Epoch::MAX;
```

Comment:
A concern that I have with this sentinel value is that we've got a wrapping add on an `Epoch` somewhere that we miss in this review, and this would get wrapped to 0.

Eventually, but reasonably soon, this goes away entirely, right? Even if it's a bit of a hack, it might be safer to just set this to some high value that we're not going to actually get to in a reasonable amount of time say `u64::MAX >> 1`.

Reply:
^ yeah, as is we should assert that imo. But my concern is we don't catch all of these and we end up with an overflow/wrap somewhere we don't catch in the review.

Reply:
we can assert on rent collector creation `self.epoch != RENT_EXEMPT_RENT_EPOCH`

Reply:
Even if we rolled over, I'm not sure how different that is than being in 'no-man's land' of using a different sentinel where incrementing it would still not be the sentinel value (max/2) but being above it. The bug of some code incorrectly incrementing when the value is u64::max would still be present, it would just produce different uncertain behavior.